src/app/components/category/category.component.ts

```diff
@@ -1,33 +1,39 @@
 import { Component, OnInit } from '@angular/core';
-import { SpotifyApiService } from 'src/app/services/spotify.service';
 import { ActivatedRoute } from '@angular/router';
+import { SpotifyApiService } from '../../services/spotify.service';
+import { LoaderService } from '../../services/loader.service';
 
 @Component({
   selector: 'app-category',
   templateUrl: './category.component.html',
-  styleUrls: ['./category.component.scss']
+  styleUrls: ['./category.component.scss'],
 })
 export class CategoryComponent implements OnInit {
   public categoryPlaylists: any = [];
   public coverImage: string;
 
   constructor(
     private spotifyService: SpotifyApiService,
-    private route: ActivatedRoute
-  ) { }
+    private route: ActivatedRoute,
+    private loader: LoaderService
+  ) {}
 
   ngOnInit() {
-    const playlist_id = this.route.snapshot.paramMap.get('id');
-    this.spotifyService.getCategoryPlaylist(playlist_id)
-      .subscribe(({ playlists }: any) => {
-        const { items } = playlists;
-        this.coverImage = items[0].images[0].url
-        this.categoryPlaylists = items;
-      });
+    this.loader.show();
+    const { id } = this.route.snapshot.params;
+    this.spotifyService.getCategoryPlaylist(id).subscribe(({ playlists }: any) => {
+      const { items } = playlists;
+      this.coverImage = this.getImage(items);
+      this.categoryPlaylists = items;
+      this.loader.hide();
+    });
   }
 
   getCoverUrl() {
     if (this.coverImage) return 'url(' + this.coverImage + ')';
   }
 
+  getImage(items: any) {
+    return items[0] && items[0].images ? items[0].images[0].url : null;
```

Comment:
Why don't you want to use `lodash.get` method? It's pretty useful. Instead of writing such checkings you can simply add the necessary path and get a value. In this case, images could be empty or something like that and you'll get an error again.